hta/common/execution_trace.py

```diff
@@ -0,0 +1,171 @@
+        and ("execution_trace|process" not in node.name)
+    )
+    for rf_id in rf_ids:
+        et_min_rf = min(rf_id, et_min_rf)
+        et_max_rf = max(rf_id, et_max_rf)
+
+    kt_min_rf, kt_max_rf = trace_df["External id"].min(), trace_df["External id"].max()
+
+    has_overlap = kt_min_rf <= et_min_rf and kt_max_rf >= et_max_rf
+
+    logging.info(f"Trace and ET have overlap = {has_overlap}")
+    logging.info(
+        f"Trace rf_ids ({kt_min_rf}, {kt_max_rf}),"
+        f"ET rf_ids ({et_min_rf}, {et_max_rf})"
+    )
+    return has_overlap
+
+
+def correlate_execution_trace(trace: Trace, rank: int, et: ExecutionGraph) -> None:
+    """Correlate the trace from a specific rank with Execution Trace object.
+
+    Args:
+        trace (Trace): Trace object loaded using `TraceAnalysis(trace_dir=trace_dir)`
+                        or other method.
+        rank (int): Rank to correlate with.
+        et (ExecutionGraph): An Execution Trace object to correlate with.
+
+    Outcome is the trace dataframe for specified rank will have a new column
+    'et_node' that includes the correlated node index in Execution Trace.
+
+    We use two different approaches depending if the PyTorch and ET trace
+        1) Have overlap: correlation is done using record function ID.
+        2) Do not have overlap: correlation is done by comparing the two
+            trees using a graph edit distance similarity algorithm.
+
+    Please note (2) is not supported yet and will come in future PRs.
+    """
+    trace_df = trace.get_trace(rank)
+
+    if not _et_has_overlap(trace_df, et):
+        logging.error(
+            "Execution Trace and kineto trace do not overlap, this mode is not currently supported"
+        )
+        return
+
+    # Mapping from rf_id to et node id
+    rf_id_to_et_node_id = {node.rf_id: id for (id, node) in et.nodes.items()}
+
+    # Only correlate specific events
+    sym_index = trace.symbol_table.get_sym_id_map()
+    sym_ids = [sym_index.get(cat) for cat in EXECUTION_TRACE_SUPPORTED_EVENTS]
+    logger.info(f"Supported event type ('cat') symbols = {sym_ids}")
+
+    row_indexer = trace_df["cat"].isin(sym_ids)
+    trace_df.loc[row_indexer, "et_node"] = trace_df.apply(
+        lambda row: rf_id_to_et_node_id.get(row["External id"], None), axis=1
+    )
+    return
+
+
+def add_et_column(trace_df: pd.DataFrame, et: ExecutionGraph, column: str) -> None:
+    """Add columns from Execution Trace nodes into the trace dataframe. Please
+    run this after running correlate_execution_trace(...).
+    Args:
+        trace_df (pd.DataFrame): Dataframe for trace from one rank. Please
+                                 run correlate_execution_trace() on the trace dataframe
+                                 first so that the `et_node` is populated..
+        et (ExecutionGraph): The Execution Trace object.
+        column (stR): Column to add from the corresponding Execution Trace node.
+    """
```

Comment:
Add `Returns: None` in the docstring. It is missing in `correlate_execution_trace` as well.